server/src/Persistence.hs

```diff
@@ -69,9 +76,94 @@ apply op val = do
 
 -- * IO
 
+loadFromBackend :: StorageBackend -> PersistenceConfig -> IO (Either String PersistentValue)
+loadFromBackend File pc = loadFile pc
+loadFromBackend Sqlite pc = loadSqliteFile pc
+
+syncToBackend :: StorageBackend -> PersistentValue -> IO ()
+syncToBackend File pv = syncFile pv
+syncToBackend Sqlite pv = syncSqliteFile pv
+
+-- * SQLite loading and syncing
+
+-- | There is currently just one row and it contains only one column of type SBS.ByteString.
+-- This single field holds the whole JSON value for now.
+data JsonRow = JsonRow {jsonByteString :: SBS.ByteString} deriving (Show)
+
+instance FromRow JsonRow where
+  fromRow = JsonRow <$> field
+
+loadSqliteFile :: PersistenceConfig -> IO (Either String PersistentValue)
+loadSqliteFile config = runExceptT $ do
+  liftIO $ forM_ (pcMetrics config) $ \m -> do
+    size <- getFileSize (pcDataFile config)
+    Metrics.setDataSize size m
+
+  value <- readSqliteData (pcDataFile config) (pcLogger config)
+  valueVar <- lift $ newTVarIO value
```

Comment:
This setup is duplicated here from `loadFile`, can we extract it so it is in one place?

Reply:
Yep, done!